maestrowf/interfaces/script/localpooladapter.py

```diff
@@ -0,0 +1,204 @@
+import logging
+import os
+
+from maestrowf.abstracts.interfaces import SchedulerScriptAdapter
+from maestrowf.abstracts.enums import JobStatusCode, State, SubmissionCode, \
+    CancelCode
+from maestrowf.interfaces.script import CancellationRecord, SubmissionRecord
+from pyaestro.utilities.executor import Executor, ExecTaskState, ExecCancel
+
+LOGGER = logging.getLogger(__name__)
+
+
+class LocalPoolAdapter(SchedulerScriptAdapter):
+    """Interface class for the flux scheduler (on Spectrum MPI)."""
+
+    key = "local_pool"
+
+    def __init__(self, **kwargs):
+        """
+        Initialize an instance of the LocalPool Adapter.
+
+        The expected keyword arguments that are expected when the Flux adapter
+        is instantiated are as follows:
+        * num_processes: The number of local workers for processing.
+
+        :param **kwargs: A dictionary with default settings for the adapter.
+        """
+        super(LocalPoolAdapter, self).__init__(**kwargs)
+        num_workers = kwargs.get("num_workers", "1")
+        self.add_batch_parameter("max_workers", num_workers)
+        self._pool = Executor(num_workers)
+        self._extension = ".lp.sh"
+
+    @property
+    def extension(self):
+        return self._extension
+
+    def get_header(self, step):
+        """
+        Generate the header present at the top of Flux execution scripts.
+
+        :param step: A StudyStep instance.
+        :returns: A string of the header based on internal batch parameters and
+                  the parameter step.
+        """
+        return f"#!{self._exec}"
+
+    def get_parallelize_command(self, procs, nodes=None, **kwargs):
```

Comment:
Should we generalize this to execute or launch command rather than making parallel special?   More just about a hook for a custom launch prefix/suffix/wrapper/etc than parallel in particular right?  One scheduler thing i can see used is big memory jobs on hpc might request a whole node just for the memory, but only run serial (e.g. srun -N1 -n1..).

Reply:
Yeah -- I want to redesign this in `pyaestro` and then use it here.